src/aap_eda/core/management/commands/purge_log_records.py

```diff
@@ -0,0 +1,158 @@
+from django.db import transaction
+
+from aap_eda.core import models
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """Purge the logs from a rulebook process."""
+
+    help = (
+        "Purge log records from rulebook processes. "
+        "If activation ids or names are not specified, "
+        "all log records older than the cutoff date will be purged."
+    )
+
+    def add_arguments(self, parser: CommandParser) -> None:
+        parser.add_argument(
+            "--activation-ids",
+            nargs="+",
+            type=int,
+            dest="activation-ids",
+            help=(
+                "Specify the activation ids which you want to clear their "
+                "records (e.g., ActivationID1 ActivationID2)"
+            ),
+        )
+        parser.add_argument(
+            "--activation-names",
+            nargs="+",
+            type=str,
+            dest="activation-names",
+            help=(
+                "Specify the activation names which you want to clear their "
+                "records (e.g., ActivationName1 ActivationName2)"
+            ),
+        )
+        parser.add_argument(
+            "--date",
+            dest="date",
+            action="store",
+            required=True,
+            help=(
+                "Purge records older than this date from the database. "
+                "The cutoff date in YYYY-MM-DD format"
+            ),
+        )
+
+    def purge_log_records(self, ids: list[int], cutoff_date: str) -> None:
+        new_instance_logs = []
+        instances = models.RulebookProcess.objects.filter(
+            activation_id__in=ids,
+        )
+
+        for instance in instances:
+            new_instance_logs.append(
+                self.clean_instance_logs(instance, cutoff_date)
+            )
+
+        new_instance_logs = [
+            item for item in new_instance_logs if item is not None
+        ]
+        if len(new_instance_logs) > 0:
+            models.RulebookProcessLog.objects.bulk_create(new_instance_logs)
+
+    def clean_instance_logs(
+        self,
+        instance: models.RulebookProcess,
+        cutoff_date: str,
+    ) -> Union[models.RulebookProcessLog, None]:
+        try:
+            ts = datetime.strptime(cutoff_date, "%Y-%m-%d")
+        except ValueError:
+            raise ValueError("Timestamp must be in 'YYYY-MM-DD' format.")
+
+        log_records = models.RulebookProcessLog.objects.filter(
+            activation_instance=instance,
+            log_timestamp__lt=int(ts.timestamp()),
+        )
+        if not log_records.exists():
+            return
+
+        log_records.delete()
+        dt = f"{datetime.now().strftime('%Y-%m-%d %H:%M:%S')}"
+
+        return models.RulebookProcessLog(
+            log=f"All log records are purged at {dt}.",
+            activation_instance_id=instance.id,
+            log_timestamp=int(datetime.now().timestamp()),
```

Comment:
Should not log_timestamp be the cutoff time or the log_timstamp of the last deleted log? This way when sorted by log_timestamp we can see the messages in logic order.